File: src/main/java/com/kingmang/lazurite/runtime/values/LzrEnum.java
package com.kingmang.lazurite.runtime.values;

import org.jetbrains.annotations.NotNull;

import java.util.Map;

public class LzrEnum implements LzrValue {

    private final Map<String, LzrString> enums;

    public LzrEnum(Map<String, LzrString> enums) {
        this.enums = enums;
    }

    public LzrString get(String enm) {
        return enums.get(enm);
    }

    @Override
    public Object raw() {
        return enums;
    }

    @Override
    public int asInt() {
        return enums.size();
    }

    @Override
    public double asNumber() {
        return enums.size();
    }

    @Override
    public String asString() {
        return enums.toString();
    }

    @Override
    public int[] asArray() {
        return new int[0];
    }

    @Override
    public int type() {
        return 0;
    }

    @Override
    public int compareTo(@NotNull LzrValue o) {
        return 0;
    }
}

File: src/main/java/com/kingmang/lazurite/runtime/values/LzrEnum.kt
package com.kingmang.lazurite.runtime.values

class LzrEnum(private val enums: Map<String, LzrString>) : LzrValue {
    fun get(enm: String): LzrString? =
        this.enums[enm]

    override fun raw(): Map<String, LzrString> =
        this.enums

    override fun asInt(): Int =
        this.enums.size

    override fun asNumber(): Double =
        this.enums.size.toDouble()

    override fun asString(): String =
        this.enums.toString()

    override fun asArray(): IntArray =
        IntArray(0)

    override fun type(): Int =
        0

    override fun compareTo(other: LzrValue): Int =
        0
}
